workers/common/src/test/kotlin/common/context/OrtServerAuthenticatorTest.kt

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.workers.common.context
+
+import io.kotest.core.spec.style.WordSpec
+import io.kotest.matchers.nulls.beNull
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+
+import io.mockk.mockk
+
+import java.net.Authenticator
+import java.net.URI
+
+class OrtServerAuthenticatorTest : WordSpec() {
+    /** Stores the original authenticator, so that it can be restored after the test. */
+    private var originalAuthenticator: Authenticator? = null
+
+    init {
+        beforeEach {
+            originalAuthenticator = Authenticator.getDefault()
+        }
+
+        afterEach {
+            Authenticator.setDefault(originalAuthenticator)
+        }
+
+        "install()" should {
+            "check whether the authenticator is already installed" {
+                val authenticator = OrtServerAuthenticator.install()
+
+                OrtServerAuthenticator.install() shouldBe authenticator
```

Comment:
How does this test work, does it check for referential equality so that calling `install` again does not create a new instance?

Reply:
Maybe make this clearer via the `shouldBeSameInstanceAs` matcher?

Reply:
Done.